src/niftyone/figures/generator.py

```diff
@@ -0,0 +1,91 @@
+"""Generator classes for different views."""
+
+import logging
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Callable, Generic, Type, TypeVar
+
+import nibabel as nib
+import pandas as pd
+from bids2table import BIDSEntities, BIDSTable
+from matplotlib.figure import Figure
+from PIL.Image import Image
+
+import niclips.image as noimg
+
+generator_registry: dict[str, Type["ViewGenerator"]] = {}
+
+T = TypeVar("T", bound="ViewGenerator")
+
+
+def register(name: str) -> Callable[[Type[T]], Type[T]]:
```

Comment:
Why `Type[T]` instead of `T`?

Reply:
`Type[T]` for working with the (sub)class directly. The registration function does this, registering the generator classes. Unless I am misunderstanding?

Reply:
I do not understand why `T` and not `Type["ViewGenerator"]`? Instances of type `T` here are instances of `ViewGenerator` or its subclasses, no?

Reply:
Just using ViewGenerator would be the non generic version. `Type[]` is not needed at all.

Reply:
Set this back to `type[T]`. 

re: @nx10's comment, had an offline discussion, largely around handling `mypy` throwing an error. Something about mypy not understanding inheritance where `type[B] != type[A]` even though `B` is a subclass of `A`. 

re: the `type["ViewGenerator"]` comment, while `mypy` is happy with this, Python isn't fully type-safe. It should probably error out here since subclasses would be instantiated as "ViewGenerator" and in theory, we shouldn't be able to call methods of the subclass. 

Hopefully that made sense, still trying to wrap my head around the second point.